src/rgw/rgw_rados.h

```diff
@@ -328,6 +329,9 @@ using tombstone_cache_t = lru_map<rgw_obj, tombstone_entry>;
 
 class RGWIndexCompletionManager;
 
+thread_local static std::minstd_rand id_rng(std::random_device{}());
```

Comment:
`minstd_rand` is fast, which is good.

it's an LCG, so people could conceivably break it, as well, but it would take several IDs to do so rather than just the one.

Reply:
Yeah, it's a fair point that someone could still possibly find the pattern in a particular thread's pseudo-random sequence, however, the fact that each thread on each daemon uses it's own random seed (and that there is no thread ID included in the request ID) would generally make that quite challenging. I believe it's an acceptable tradeoff for the performance.

Reply:
is there a reason not to use the existing thread-local generators from `include/random.h`?

`ceph::util::generate_random_number<uint64_t>()` uses a thread-local `std::default_random_engine`, which is already being seeded on these same rgw threads. i'd prefer not to drain more entropy from `std::random_device` unless `std::default_random_engine` really isn't suitable here

Reply:
Sounds good to me. Just don't use the mutex versions.